### ISSUE TITLE
Building data doc to S3 prevents static bucket website endpoint from loading

### ISSUE THREAD

**Issue by Andrew088** [fivetran/great_expectations#2910]
**Describe the bug**
When building data docs to S3, with properly configured data context, the bucket URL does not seem to receive any data from the index.html object. However, if you were to manually upload the data_docs directory files to S3, the bucket URL loads.

**To Reproduce**
1. Add the following configuration under `data_docs_sites` in your `great_expectations.yml` file:
```yaml
s3_site: # this is a user-selected name - you may select your own
  class_name: SiteBuilder
  store_backend:
    class_name: TupleS3StoreBackend
    bucket: <bucket-name>  # UPDATE the bucket name here to match the bucket you configured above.
  site_index_builder:
    class_name: DefaultSiteIndexBuilder
    show_cta_footer: true
```
2. Run validations on your data
3. Run the command in terminal to build data docs:
```bash
great_expectations --v3-api -y docs build --site-name s3_site
```
4. Go to the S3 bucket you uploaded the data docs 
5. Click on the properties tab, scroll all the way down, and edit Static Website Hosting 
6. Enable static website hosting, set `index.html` as the index document, and then save changes
7. Scroll all the way down again and go to the link under Bucket website endpoint

**Expected behavior**
This should direct you to your Data Docs

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: [e.g. 0.13.19]

**Additional context**
I believe this issue is a result of great_expectations setting the S3 object's `Content-Encoding` metadata to "utf-8" when running the build data docs command. I don't think content-encoding is being used correctly here, as utf-8 is a character set and not a a data compression algorithm.

I have applied a quick fix to the issue on my current fork of version 0.13.19 by removing `ContentEncoding=content_encoding,` from: https://github.com/great-expectations/great_expectations/blob/0c01c2fe4c59383d7dc350cc005d6056bf531c09/great_expectations/data_context/store/tuple_store_backend.py#L555-L559

This seemed to have fixed the issue of the bucket endpoint not loading the index.html properly. 

I have noticed this pattern across other classes, and I think there should be a proper fix for this issue.
Thank you so much for your time, and I look forward to hearing your thoughts/feedback on this issue.

**Comment by talagluck** [comment]
Thanks for raising this issue @Andrew088 ! Is this something that might be addressed by or related to #2967? Would you be interested in attempting a contribution here? I'm happy to offer review, or any additional guidance that might be helpful.

**Comment by talagluck** [comment]
Hi @Andrew088 - I'm going to close this for now, but please feel free to re-open if you feel this is still an issue, or are interested in making a contribution. Thanks!
